e2e/src/test/scala/scalapb/zio_grpc/ClientCallsSpec.scala

```diff
@@ -4,34 +4,40 @@ import io.grpc.{CallOptions, ManagedChannelBuilder}
 import scalapb.zio_grpc.TestUtils._
 import scalapb.zio_grpc.client.ClientCalls
 import scalapb.zio_grpc.testservice._
-import zio.Schedule
-import zio.test.Assertion._
+import zio._
 import zio.test._
+import zio.test.Assertion._
 
 object ClientCallsSpec extends ZIOSpecDefault {
 
   def unarySuite =
     suite("unaryCall")(
       test("should not fail with 'INTERNAL: already started' on retry") {
         for {
-          meta <- SafeMetadata.make
-          res  <- ClientCalls
-                    .unaryCall(
-                      // This test does not require working server
-                      new ZChannel(
-                        ManagedChannelBuilder
-                          .forAddress("localhost", 0)
-                          .usePlaintext()
-                          .build(),
-                        Nil
-                      ),
-                      scalapb.zio_grpc.testservice.TestServiceGrpc.METHOD_UNARY,
-                      CallOptions.DEFAULT,
-                      meta,
-                      Request(Request.Scenario.DELAY, in = 12)
-                    )
-                    .retry(Schedule.recurs(2))
-                    .exit
+          channel <- ZIO
+                       .acquireRelease(
```

Comment:
Is there something we can factor out of this into the companion object of ZChannel to make it easier for the users to create a scoped channel?

Reply:
Sure, let me work this out.

Reply:
@thesamet done